File: src/algorithms/hamming-weight.js
/**
 * Takes in the binary representation of an unsigned integer and returns the
 * number of '1' bits it has.
 *
 * @format
 * @param {number} n - A binary string of length 32
 * @returns {number} - Returns the count of '1' bits in n
 */

function hammingWeight(n) {
  let count = 0;
  while (n !== 0) {
    n &= n - 1;
    count++;
  }
  return count;
}

File: src/algorithms/hamming-weight.py
def hammingWeight(n):
    """Given a positive integer n, write a function that returns the number of 
    set bits in its binary representation (also known as the Hamming weight).

    Args:
        n (int): A positive integer.

    Returns:
        int: Returns the hamming weight of n.
    """
    count = 0
    n = bin(n)[2:]
    for i in range(len(n)):
        if n[i] == "1":
            count += 1
    return count
